osbenchmark/utils/opts.py

```diff
@@ -127,13 +128,19 @@ def __init__(self, option_strings, dest, nargs=None, **kwargs):
         super().__init__(option_strings, dest, nargs=nargs, **kwargs)
 
     def __call__(self, parser, namespace, values, option_string=None):
-        my_dict = {}
+        custom_dict = {}
         for kv in values:
-            k,v = kv.split(":")
-            my_dict[k] = v
-        setattr(namespace, self.dest, my_dict)
-
-        return my_dict
+            try:
+                k,v = kv.split(":")
+                custom_dict[k] = v
+            except ValueError:
+                raise exceptions.InvalidSyntax(
+                    "StoreKeyPairAsDict: Could not convert string to dict due to invalid syntax. If using parameter " +
+                    "--number-of-docs, ensure input follows format: " + "<index1>:<doc_count1> <index2>:<doc_count2> ... "
```

Comment:
This is helpful, but it ties the message to the --number-of-docs option.  If this action gets re-used for some other option, admittedly unlikely for now, it might need to be made more general.

Reply:
This came to mind when creating this message. I'll simplify it and make it more generic.